### ISSUE TITLE
wai-session-0.3.1 does not compile

### ISSUE THREAD

**Issue by peti** [singpolyma/wai-session#8]
Citing from http://hydra.cryp.to/build/933776/nixlog/3/raw:

~~~~
Network/Wai/Session/Map.hs:28:9:
    Non type-variable argument
      in the constraint: HasUpdate
                           t (Map k (StateVar (Map k v))) (Map k (StateVar (Map k v)))
    (Use FlexibleContexts to permit this)
    When checking that ‘mapStore'’ has the inferred type
      mapStore' :: forall k v (m :: * -> *) (m1 :: * -> *) t k1.
                   (Monad m1, Ord k, Ord k1, MonadIO m,
                    HasUpdate
                      t (Map k1 (StateVar (Map k v))) (Map k1 (StateVar (Map k v))),
                    HasGetter t (Map k1 (StateVar (Map k v)))) =>
                   IO k1 -> t -> Maybe k1 -> IO (Session m k v, m1 k1)
    In an equation for ‘mapStore’:
        mapStore gen
          = liftM (mapStore' gen) (newThreadSafeStateVar Map.empty)
          where
              mapStore' _ ssv (Just k)
                = do { m <- get ssv;
                       .... }
              mapStore' genNewKey ssv Nothing
                = do { newKey <- genNewKey;
                       .... }
~~~~~~

**Comment by creichert** [comment]
This is fixed in #7 which has not been merged yet.

**Comment by peti** [comment]
It looks like this package is no longer maintained. :disappointed:

**Comment by athanclark** [comment]
:( any other leads, @peti?

**Comment by singpolyma** [comment]
Hey all, sorry for the delays on this.  Looking into it now

**Comment by singpolyma** [comment]
So, this is a bug specific to GHC 7.10 ?

**Comment by creichert** [comment]
@singpolyma Yea, enabling that extension is required more often for type inference in GHC 7.10. Enabling it has worked fine for me.

https://ghc.haskell.org/trac/ghc/wiki/Migration/7.10#Inferredtype-signaturesnowmayrequiretoenableFlexibleContextsGADTsorTypeFamilies

**Comment by singpolyma** [comment]
I've merged the pull request, but this sort of thing makes me pretty unhappy -- I need to declare a language extension just for inferred types now?  Makes it much harder to keep my code clean :P

**Comment by singpolyma** [comment]
I've pushed the fix to hackage as 0.3.2, for people on the cutting edge compilers :)

**Comment by athanclark** [comment]
Thank you!!

On Sat, Jul 4, 2015 at 2:54 PM, Stephen Paul Weber <notifications@github.com
